src/proto/media/file/dir.rs

```diff
@@ -20,25 +20,20 @@ impl Directory {
         Self(RegularFile::new(handle))
     }
 
-    /// Read the next directory entry
+    /// Read the next directory entry.
     ///
-    /// Try to read the next directory entry into `buffer`. If the buffer is too small, report the
-    /// required buffer size as part of the error. If there are no more directory entries, return
-    /// an empty optional.
+    /// Under the hood, this wraps [`super::RegularFile::read`]. Please refer to it's documentation.
     ///
     /// The input buffer must be correctly aligned for a `FileInfo`. You can query the required
     /// alignment through the `Align` trait (`<FileInfo as Align>::alignment()`).
     ///
     /// # Arguments
     /// * `buffer`  The target buffer of the read operation
     ///
-    /// # Errors
-    /// * `uefi::Status::NO_MEDIA`           The device has no media
-    /// * `uefi::Status::DEVICE_ERROR`       The device reported an error, the file was deleted,
-    ///                                      or the end of the file was reached before the `read()`.
-    /// * `uefi::Status::VOLUME_CORRUPTED`   The filesystem structures are corrupted
-    /// * `uefi::Status::BUFFER_TOO_SMALL`   The buffer is too small to hold a directory entry,
-    ///                                      the required buffer size is provided into the error.
+    /// # Return Value
+    /// This method returns a UEFI [`Result`]. In the success case, this method returns an `Option`
```

Comment:
Maybe I should drop `UEFI` in UEFI result. Sounds confusing. Under the hood, it is a regular Rust result.

Reply:
Since the `uefi::Result` is used almost everywhere in this crate, I don't think it needs to be explicitly called out here. However, we probably should add something to the top-level crate doc (https://docs.rs/uefi/latest/uefi/) to briefly summarize how `uefi::Result` is defined and used.

Reply:
Obsolete